tests/scene/test_visual_shader.h

```diff
@@ -0,0 +1,198 @@
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef TEST_VISUAL_SHADER_H
+#define TEST_VISUAL_SHADER_H
+
+#include "scene/resources/visual_shader.h"
+
+#include "tests/test_macros.h"
+
+namespace TestVisualArray {
+
+TEST_CASE("[SceneTree][VisualShader] Object creation and parameter") {
+	Ref<VisualShader> vs = memnew(VisualShader);
+	CHECK(vs.is_valid());
+
+	CHECK(vs->get_mode() == Shader::MODE_SPATIAL);
+	vs->set_mode(Shader::MODE_PARTICLES);
+	CHECK(vs->get_mode() == Shader::MODE_PARTICLES);
+
+	vs->set_mode(Shader::MODE_FOG);
+	CHECK(vs->get_mode() == Shader::MODE_FOG);
+
+	vs->set_mode(Shader::MODE_PARTICLES);
+	CHECK(vs->get_mode() == Shader::MODE_PARTICLES);
+}
+
+TEST_CASE("[SceneTree][VisualShader] Testing VisualShaderNodes") {
+	SUBCASE("Testing Node Creation") {
+		Ref<VisualShader> vs = memnew(VisualShader);
+		CHECK(vs.is_valid());
+
+		Ref<VisualShaderNode> vsn1 = memnew(VisualShaderNodeInput);
+		CHECK(vsn1.is_valid());
+		vs->add_node(VisualShader::TYPE_COLLIDE, vsn1, Vector2(0, 0), 3);
+		CHECK(vs->get_node(VisualShader::TYPE_COLLIDE, 3) == vsn1);
+
+		Ref<VisualShaderNode> vsn2 = memnew(VisualShaderNodeInput);
+		CHECK(vsn2.is_valid());
+		vs->add_node(VisualShader::TYPE_FOG, vsn2, Vector2(1, 2), 4);
+		CHECK(vs->get_node(VisualShader::TYPE_FOG, 4) == vsn2);
+
+		Ref<VisualShaderNode> vsn3 = memnew(VisualShaderNodeInput);
+		CHECK(vsn3.is_valid());
+		vs->add_node(VisualShader::TYPE_LIGHT, vsn3, Vector2(4, 2), 5);
+		CHECK(vs->get_node(VisualShader::TYPE_LIGHT, 5) == vsn3);
+
+		Ref<VisualShaderNode> vsn4 = memnew(VisualShaderNodeInput);
+		CHECK(vsn4.is_valid());
+		vs->add_node(VisualShader::TYPE_SKY, vsn4, Vector2(1, 10), 6);
+		CHECK(vs->get_node(VisualShader::TYPE_SKY, 6) == vsn4);
+
+		ERR_PRINT_OFF;
+
+		// Testing for Invalid entries
+		Ref<VisualShaderNode> vsn5 = memnew(VisualShaderNodeInput);
+		CHECK(vsn5.is_valid());
+		vs->add_node(VisualShader::TYPE_SKY, vsn5, Vector2(1, 10), 0);
+		CHECK_FALSE(vs->get_node(VisualShader::TYPE_SKY, 0) == vsn4);
+		Ref<VisualShaderNode> vsn6 = memnew(VisualShaderNodeInput);
```

Comment:
```suggestion
		CHECK_FALSE(vs->get_node(VisualShader::TYPE_SKY, 0) == vsn4);
		
		Ref<VisualShaderNode> vsn6 = memnew(VisualShaderNodeInput);
```
add new line here...

Reply:
Done 👍